tests/ha/test_multi_pod_failure.py

```diff
@@ -2341,3 +2342,386 @@ def test_degraded_mpu_after_kpods_fail(self):
+                        count, pod_name)
+
+            LOGGER.info("Step 3: Check cluster status")
+            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+            assert_utils.assert_false(resp[0], resp)
+            LOGGER.info("Step 3: Cluster is in degraded state")
+
+            LOGGER.info("Step 4: Check services status that were running on pods which are deleted")
+            hostname = self.pod_dict.get(pod_name)[0]
+            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod_name], fail=True,
+                                                               hostname=hostname)
+            assert_utils.assert_true(resp[0], resp[1])
+            pod_list.remove(pod_name)
+            LOGGER.info("Step 4: Services of pods are in offline state")
+
+            LOGGER.info("Step 5: Check services status on remaining pods %s", pod_list)
+            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=pod_list, fail=False)
+            LOGGER.debug("Response: %s", resp)
+            assert_utils.assert_true(resp[0], resp)
+            LOGGER.info("Step 5: Services of remaining pods are in online state.")
+
+            uploading_parts = random.sample(parts, 10)
+            LOGGER.info("Step 6: Perform partial multipart upload for %s parts out of total %s",
+                        uploading_parts, len(parts))
+            resp = self.ha_obj.partial_multipart_upload(s3_data=self.s3_clean,
+                                                        bucket_name=self.bucket_name,
+                                                        object_name=self.object_name,
+                                                        part_numbers=uploading_parts,
+                                                        remaining_upload=True, mpu_id=mpu_id,
+                                                        multipart_obj_size=file_size,
+                                                        total_parts=total_parts,
+                                                        multipart_obj_path=object_path)
+
+            assert_utils.assert_true(resp[0], f"Failed to upload parts {resp[1]}")
+            LOGGER.info("Listing parts of partial multipart upload")
+            res = s3_mp_test_obj.list_parts(mpu_id, self.bucket_name, self.object_name)
+            assert_utils.assert_true(res[0], res)
+            for part_n in res[1]["Parts"]:
+                assert_utils.assert_list_item(uploading_parts, part_n["PartNumber"])
+            LOGGER.info("Listed parts of partial multipart upload: %s", res[1])
+            parts_etags.extend(resp[3])
+            parts = [ele for ele in parts if ele not in uploading_parts]
+            LOGGER.info("Step 6: Successfully performed partial multipart upload for %s parts out "
+                        "of total %s", uploading_parts, len(parts))
+
+        LOGGER.info("Step 7: Upload remaining %s parts", parts)
```

Comment:
Same as above.

Reply:
> Same as above.

Updated logic for total_parts = self.kvalue * 10 + 50